packages/hackmcx-api/src/controllers/users.controller.js

```diff
@@ -46,21 +44,21 @@ export async function postUser(req, res){
         res.send({error: "Last Name cannot be missing or blank."});
         return
     }
-    if(usernameExists){
-        res.statusCode = 400;
-        res.send({error: "Invalid Username: Username is already taken."});
-        return
-    }
     if(!noGivenImageUrl && notPicture){
         res.statusCode = 400;
-        res.send({error: "Invalid image URL: Image url is either missing or invalid."});
+        res.send({error: "Invalid image URL: Image url is invalid."});
         return
     }
 
-    let id =  noGivenImageUrl ? 
-                (await dbClient.table('users').insert({username: req.body.username, first_name: req.body.firstname, last_name: req.body.lastname }))[0] :
-                (await dbClient.table('users').insert({username: req.body.username, first_name: req.body.firstname, last_name: req.body.lastname, imageUrl: req.body.imageUrl}))[0]
-    res.statusCode = 201;
-    res.header('Location',`${req.baseUrl}/${id}` );
-    res.send();
+    try{
+        let id =  noGivenImageUrl ? 
+                    (await dbClient.table('users').insert({username: req.body.username, first_name: req.body.firstname, last_name: req.body.lastname }))[0] :
+                    (await dbClient.table('users').insert({username: req.body.username, first_name: req.body.firstname, last_name: req.body.lastname, imageUrl: req.body.imageUrl}))[0]
+        res.statusCode = 201;
+        res.header('Location',`${req.baseUrl}/${id}` );
+        res.send();
+    }catch(e){
+        res.statusCode = 400;
```

Comment:
We can't just assume any error is a PK conflict, we might want to check the error message to make sure in this case, and return a 400 if there is a PK conflict, and maybe something different otherwise. 

Eventually we might want to look at a global error handler to allow the app to return 5XX errors instead of crashing.